api_tests/registrations/views/test_registration_detail.py

```diff
@@ -1,297 +1,293 @@
-
-        self.user_two = AuthUserFactory()
-        self.user_three = AuthUserFactory()
-
-        self.registration_approval = RegistrationApprovalFactory(state='unapproved', approve=False, user=self.user)
-        self.unapproved_registration = Registration.find_one(Q('registration_approval', 'eq', self.registration_approval))
-        self.unapproved_url = '/{}registrations/{}/'.format(API_BASE, self.unapproved_registration._id)
-
-        self.public_project = ProjectFactory(title="Project One", is_public=True, creator=self.user)
-        self.private_project = ProjectFactory(title="Project Two", is_public=False, creator=self.user)
-        self.public_registration = RegistrationFactory(project=self.public_project, creator=self.user, is_public=True)
-        self.private_registration = RegistrationFactory(project=self.private_project, creator=self.user)
-        self.public_url = '/{}registrations/{}/'.format(API_BASE, self.public_registration._id)
-        self.private_url = '/{}registrations/{}/'.format(API_BASE, self.private_registration._id)
-
-        self.private_registration.add_contributor(self.user_two, permissions=[permissions.READ])
-        self.private_registration.add_contributor(self.user_three, permissions=[permissions.WRITE])
-        self.private_registration.save()
-
-        self.payload = {
-            "data": {
-                "id": self.private_registration._id,
-                "type": "registrations",
-                "attributes": {
-                    "public": True,
+        node_url = '/{}nodes/{}/'.format(API_BASE, private_project._id)
+        res = app.get(node_url, auth=user.auth)
+        assert res.status_code == 200
+        assert 'registrations' in res.json['data']['relationships']
+
+        res = app.get(private_url, auth=user.auth)
+        assert res.status_code == 200
+        assert 'registrations' not in res.json['data']['relationships']
+
+@pytest.mark.django_db
+class TestRegistrationUpdate:
+
+    @pytest.fixture()
+    def read_only_contributor(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def read_write_contributor(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def registration_approval(self, user):
+        return RegistrationApprovalFactory(state='unapproved', approve=False, user=user)
+
+    @pytest.fixture()
+    def unapproved_registration(self, registration_approval):
+        return Registration.find_one(Q('registration_approval', 'eq', registration_approval))
+
+    @pytest.fixture()
+    def unapproved_url(self, unapproved_registration):
+        return '/{}registrations/{}/'.format(API_BASE, unapproved_registration._id)
+
+    @pytest.fixture()
+    def public_project(self, user):
+        return ProjectFactory(title='Project One', is_public=True, creator=user)
+
+    @pytest.fixture()
+    def private_project(self, user):
+        return ProjectFactory(title='Project Two', is_public=False, creator=user)
```

Comment:
The title's aren't necessary but can help when debugging tests. Better to make them more descriptive, though--e.g. `Private Project`.